From c6ecc6b823e556f0cea0759e82a7509ed32b91a1 Mon Sep 17 00:00:00 2001
From: Aleksandrs Vinarskis <alex.vinarskis@gmail.com>
Date: Sat, 28 Dec 2024 22:55:02 +0100
Subject: [PATCH 27/39] UBUNTU: SAUCE: arm64: dts: qcom:
 x1e80100-acer-swift-sf14-11: Enable HBR3 on external DPs

BugLink: https://bugs.launchpad.net/bugs/2100858

When no link frequencies are set, msm/dp driver defaults to HBR2 speed.
Explicitly list supported frequencies including HBR3/8.1Gbps for all
external DisplayPort(s).

Signed-off-by: Aleksandrs Vinarskis <alex.vinarskis@gmail.com>
Signed-off-by: Tobias Heider <tobias.heider@canonical.com>
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 arch/arm64/boot/dts/qcom/x1p64100-acer-swift-sf14-11.dts | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/x1p64100-acer-swift-sf14-11.dts b/arch/arm64/boot/dts/qcom/x1p64100-acer-swift-sf14-11.dts
index 7d59a2ff9e94..b81ebb659d0e 100644
--- a/arch/arm64/boot/dts/qcom/x1p64100-acer-swift-sf14-11.dts
+++ b/arch/arm64/boot/dts/qcom/x1p64100-acer-swift-sf14-11.dts
@@ -816,6 +816,7 @@ &mdss_dp0 {
 
 &mdss_dp0_out {
 	data-lanes = <0 1>;
+	link-frequencies = /bits/ 64 <1620000000 2700000000 5400000000 8100000000>;
 };
 
 &mdss_dp1 {
@@ -824,6 +825,7 @@ &mdss_dp1 {
 
 &mdss_dp1_out {
 	data-lanes = <0 1>;
+	link-frequencies = /bits/ 64 <1620000000 2700000000 5400000000 8100000000>;
 };
 
 &mdss_dp3 {
-- 
2.48.1

